src/components/AddComponent/AuthTokenModal.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import {
+  Alert,
+  AlertVariant,
+  Button,
+  ButtonType,
+  ButtonVariant,
+  Form,
+  ModalVariant,
+  Stack,
+  StackItem,
+  Text,
+  TextContent,
+  TextInputTypes,
+  TextVariants,
+} from '@patternfly/react-core';
+import { Formik, FormikValues } from 'formik';
+import { useChrome } from '@redhat-cloud-services/frontend-components/useChrome';
+import { InputField } from '../../shared';
+import { ComponentProps, createModalLauncher } from '../modal/createModalLauncher';
+import { useNamespace } from '../NamespacedPage/NamespacedPage';
+
+type AuthTokenModalProps = ComponentProps & {
+  accessTokenName: string;
+};
+
+export const AuthTokenModal: React.FC<AuthTokenModalProps> = ({ onClose, accessTokenName }) => {
+  const [error, setError] = React.useState();
+  const { namespace } = useNamespace();
+  const {
+    auth: { getToken },
+  } = useChrome();
+
+  const submitToken = React.useCallback(
+    async (values: FormikValues) => {
+      try {
+        const token = await getToken();
+        await fetch(
+          `https://spi-oauth-route-spi-system.apps.appstudio-stage.x99m.p1.openshiftapps.com/token/${namespace}/${accessTokenName}`,
+          {
+            method: 'POST',
+            credentials: 'include',
+            headers: {
+              Authorization: `Bearer ${token}`,
+            },
+            body: JSON.stringify({
+              username: values.username,
+              // eslint-disable-next-line camelcase
+              access_token: values.token,
+            }),
+          },
+        );
```

Comment:
you can encapsulate this in a hook which return spiUtils
something like 
```
const {uploadToken} = useSpiAPI()
uploadToken(accessTokenName);
```

Reply:
Updated.